salt/metalk8s/volumes/cleanup.sls

```diff
@@ -0,0 +1,56 @@
+# Clean already attached loop devices before applying highstate, to avoid
+# creating duplicates for a same sparse file.
+# This state can cleanup either "manually", for upgrade, or by disabling the
+# systemd units, for downgrade.
+{%- set target_version = pillar.metalk8s.nodes[grains.id].version %}
+
+include:
+  - .installed
```

Comment:
Hmm do we really need to include this "installed" when going to version < 2.7.0 ? Because it's weird to me, we want to remove the systemd unit file but first, we ensure the unit file exists

Reply:
`.installed` only takes care of "dependencies": some packages for formatting devices, but most importantly, it ensures the cleanup script and the systemd unit **template** are available.

When going from 2.7 to 2.6, we only need the template, to make sure the service can be found (not sure how `service.dead` behaves if this file does not exist).
When going from 2.6 to 2.7, we need the cleanup script, which we call manually.

Reply:
> When going from 2.7 to 2.6, we only need the template, to make sure the service can be found (not sure how service.dead behaves if this file does not exist).

```
local:
----------
          ID: toto
    Function: service.dead
      Result: True
     Comment: The named service toto is not available
     Started: 13:39:24.705098
    Duration: 26.128 ms
     Changes:   

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:  26.128 ms
```

I got your point but I think if we have the service running then we already have all the dependencies no ?
Anyway, as you want, it's not really important we can include this "installed" it does not hurt

Reply:
I still need it when upgrading, so maybe I can only include `if target_version >= 2.7.0`?

Reply:
Yes maybe the good approach, I think

Reply:
Done